src/main/java/io/zipcoder/casino/GameClasses/DiceGames/OU7.java

```diff
@@ -1,40 +1,98 @@
+    OU7Player ou7Player1;
+    Integer gameBalance;
+    Integer betAmount;
+    String actualGuess;
 
-        return null;
+    // ***************************************************************************** StartGame()  - game logic *********
+    public void startGame(Membership membership) {
+
+        playerBuilder(membership);
+        Boolean keepPlaying = true;
+
+        while (keepPlaying.equals(true)) {
+            System.out.println("\n\n" + "            Welcome to Over/Under 7             : " + ou7Player1.getName() + "\n");
+            System.out.print           ("      Enter your bet type - Over/Under 7        : ");
+            Scanner playerInput = new Scanner(System.in);
+            String playerGuess = playerInput.nextLine();
+            Boolean guess = evaluateGuessToRoll(pairOfDice, playerGuess);
+
+            keepPlaying = playAgain();
+        }
     }
 
-    public Boolean win() {
+    // ***************************************************************************** Build Player **********************
+    public Player playerBuilder(Membership accountDetails) {
 
-        return null;
+        ou7Player1 = new OU7Player(accountDetails);
+        gameBalance = ou7Player1.getBalance();
+
+        return ou7Player1;
     }
+    // ***************************************************************************** Evaluate Guess To Roll ************
+    public Boolean evaluateGuessToRoll(Dice rolledDice, String actualGuess) {
 
-    public Boolean playAgain() {
-        return null;
+        Integer thisTurnsRoll = rolledDice.rollAndSum(pairOfDice);
+        //                 "      Enter your bet type - Over/Under 7        : "
+        System.out.println("                  Your Roll                     : " + thisTurnsRoll);
+        if (thisTurnsRoll > 7 && actualGuess.equals("Over") || actualGuess.equals("over")) {
+            System.out.println("Congrats! You win");
+            // gameBalance += betAmount;
+            return true;
+        } else if (thisTurnsRoll < 7 && actualGuess.equals("Under") ||  actualGuess.equals("under")){
+            // gameBalance += betAmount;
+            System.out.println("Congrats! You win");
+            return true;
+//                  }else if(actualGuess.equals("Equals")&&thisTurnsRoll==7){
+//                  return true;
+        } else {
+            // gameBalance -= betAmount;
+            System.out.println("Sorry. You lose");
+            return false;
+        }
     }
 
-    public Player playerBuilder() {
+    // ***************************************************************************** Play Again   **********************
+    public Boolean playAgain(){
+
+        boolean answer = false;
+        System.out.println("Do you want to play again? Y/N");
+        Scanner in =  new Scanner(System.in);
+        String res = in.nextLine().toUpperCase();
+        if (res == "Y" || res == "y")
+            answer = true;
+        else if (res == "n" || res == "N"){
```

Comment:
Our static analysis engine detects that your pull request includes a change from a Comparison of String objects using `==` or `!=`. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq), unless both strings are either constants in a source file, or have been interned using the `String.intern()` method, the same string value may be represented by two different String objects. Consider using the `equals(Object)` method instead.